lib/python/picongpu/picmi/lasers/twts_laser.py

```diff
@@ -0,0 +1,136 @@
+"""
+This file is part of PIConGPU.
+Copyright 2025 PIConGPU contributors
+Authors: Julian Lenz, Alexander Debus
+License: GPLv3+
+"""
+
+import math
+import typing
+
+import typeguard
+
+from ...pypicongpu import laser
+from ..copy_attributes import default_converts_to
+from .base_laser import BaseLaser
+from .polarization_type import PolarizationType
+
+from .. import constants
+
+
+@default_converts_to(laser.TWTSLaser)
+@typeguard.typechecked
+class TWTSLaser(BaseLaser):
+    """
+    Specifies a TWTS laser
+
+    Parameters
+    ----------
+    - wavelength: float
+        Central wavelength of the laser [m].
+
+    - waist : float
+        Spot size (1/e^2 radius) of the laser at focus [m].
+
+    - duration: float
+        Duration of the TWTS pulse [s], defined as 1 sigma of std. Gauss for intensity.
+
+    - phi: float
+        Laser incident angle [rad]
+        denoting the mean laser phase propagation direction
+        with respect to the y-axis.
```

Comment:
magical axis definition, either requires a fixed coordinate system or should be defined relative to something like the focus movement direction.
Please remember that we may not assume that the focus moves parallel to the y-direction in PICMI, since this might not be true in other PIC-codes.